frontend/src/components/FacilityBooking/ViewSection.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { DailyContainer, MainContainer } from './BlockStyles'
+import ViewBlock from './ViewBlock'
+import { RootState } from '../../store/types'
+import HourBlocks from './HourBlocks'
+import CurrentTimeLine, { isToday } from './CurrentTimeLine'
+import { getTimeBlocks } from '../../store/facilityBooking/action'
+
+export function setDefaultTimePosition(timePosition: number, arrLen: number) {
+  let defaultTimePosition = timePosition
+  if (defaultTimePosition < 0) {
+    defaultTimePosition = 0
+  }
+  if (defaultTimePosition >= arrLen) {
+    defaultTimePosition = arrLen - 1
+  }
+  return defaultTimePosition
+}
+
+const ViewScheduleBlock = () => {
+  const { timeBlocks } = useSelector((state: RootState) => state.facilityBooking)
+  const dispatch = useDispatch()
+  let defaultTimePosition = 16 //4pm
+
+  useEffect(() => {
+    dispatch(getTimeBlocks())
+    defaultTimePosition = setDefaultTimePosition(defaultTimePosition, timeBlocks.length)
```

Comment:
Not very sure why this is necessary. In case someone add a nonsense value to line 25?

Reply:
yup, also bc im not sure if there would be changes to the timeblocks length so i left it as this to prevent hardcoded value; i could leave as comment to remind them next time also?